test/misc.jl

```diff
@@ -7,3 +7,12 @@ using Pkg
     f() = Pkg.Types.UNREGISTERED_STDLIBS
     @inferred f()
 end
+
+@testset "hashing" begin
+    @test hash(Pkg.Types.Project()) == 0xd4b8cda960e4de88
```

Comment:
I don't think these should be added. It doesn't make sense to compare a hash value like this imo. You could try `hash(Pkg.Types.Project()) == hash(Pkg.Types.Project())` or something?